### PR TITLE
[1.0] fix ci

### PR DESCRIPTION
**Pull request by kolyshkin**
_TL;DR: backports of #3112 and #3069 to release-1.0 branch to fix CI broken by upstream updates._

### 1. libct/int/checkpoint_test: fix ParentImage

The ParentImage set by the test should be a path relative to
ImagesDirectory, pointing to a parent images directory (created
by pre-dump).

The parent directory is created by TempDir and so its name is not
constant but has a variable suffix. So, the config was pointing to
a non-existent directory.

This left unnoticed by criu as it assumed the parent image does not
exist, and performed a full dump.

Since criu PR https://github.com/checkpoint-restore/criu/pull/1403 (will be a part of criu 3.16) that is no longer the
case -- the invalid parent path is treated as an error, and so our
test fails like this:

```
== RUN   TestCheckpoint
    checkpoint_test.go:145: === /tmp/criu070876105/dump.log ===
    checkpoint_test.go:145: open /tmp/criu070876105/dump.log: no such file or directory
    checkpoint_test.go:146: criu failed: type DUMP errno 56
        log file: /tmp/criu070876105/dump.log
--- FAIL: TestCheckpoint (0.26s)
```

Fix this by using the actual name of the parent image dir.

This fixes release-1.0 CI against criu 3.16.

Fixes: 98f004182b9a23245
Fixes: #3278

NOTE
* this is a clean backport of #3112 to release-1.0
* in master, this was fixed implicitly by https://github.com/opencontainers/runc/pull/3100/commits/3bc606e9d30e5e616e4602bd5b1b1610cc3b8fbe

### 2. tests/int/dev: add CAP_SYSLOG to /dev/kmsg tests

Add CAP_SYSLOG to ensure that /dev/kmsg can be accesses on systems where
the sysctl kernel.dmesg_restrict = 1.

(This is a backport of #3069 to release-1.0)

### CODE REVIEW

**Comment by kolyshkin** [comment]
```
not ok 41 runc run [device cgroup allow rw char device]
# (in test file tests/integration/dev.bats, line 87)
#   `[ "$status" -eq 0 ]' failed
# runc spec (status=0):
# 
# runc run -d --console-socket /tmp/bats-run-15350/runc.61S5q1/tty/sock test_allow_char (status=0):
# 
# runc exec test_allow_char sh -c hostname | tee /dev/kmsg (status=0):
# myhostname
# runc exec test_allow_char sh -c head -n 1 /dev/kmsg (status=1):
# head: /dev/kmsg: Operation not permitted
```

This needs #3069 backported. Will add.

**Comment by kolyshkin** [comment]
OK, now I know why we need periodical (rather than per-PR) CI jobs. Will add.

**Review by mrunalp — APPROVED** [review]

**Comment by kolyshkin** [comment]
@AkihiroSuda @thaJeztah PTAL

**Review by thaJeztah — APPROVED** [review]
LGTM
